src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -625,15 +641,15 @@ evalELam :: Knowledge -> ICtx -> EvalType -> (Symbol, Sort) -> Expr -> EvalST (E
 evalELam γ ctx et (x, s) e = do
     oldPendingUnfoldings <- gets evPendingUnfoldings
     oldEqs <- gets evNewEqualities
-    (e', fe) <- eval (γ { knLams = (x, s) : knLams γ }) ctx et e
+    (e', fe) <- eval (γ { knLams = (x, s) : knLams γ }) ctx et False e
     let e2' = simplify γ ctx e'
         elam = ELam (x, s) e
     -- Discard the old equalities which miss the lambda binding
     modify $ \st -> st
       { evPendingUnfoldings = oldPendingUnfoldings
       , evNewEqualities = S.insert (elam, ELam (x, s) e2') oldEqs
       }
-    return (elam, fe)
+    return (ELam (x, s) e2', fe)
```

Comment:
No it shouldn't be needed, I'll revert the change

Reply:
But I need to test it before

Reply:
No It's actually needed, since now we need to keep track that the body was rewritten

Reply:
In that case, I suggest to use the unsimplified `e'` instead of `e2'`.

Reply:
I agree that we can miss potential rewrites at each selection. But, again, since this is the first real work (after the initial old attempt) to make reasoning with lambdas work, and there is an example that actually needs it, I suggest we accept.

Reply:
I replaced it with e'